### PR TITLE
fix & improve `eris init`, alias to `update`

### LINKED ISSUE #189: `eris update`
update the tool itself. 

can add build from source if Go installed now. binary updates will be blocking until that pipeline is fully completed.

### PR DESCRIPTION
**Pull request by zramsay**
closes #189, has a little of #203 
cleaning:
- [x] debug statements
- [x] deduplicate error messages & use some `logger.Debug`

### CODE REVIEW

File: commands/update.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	out3, err := exec.Command("go", goArgs...).CombinedOutput()
+	if err != nil {
+		fmt.Printf("error with go install ./cmd/eris:\n%s\n", string(out3))
+		os.Exit(1)
+	}
+
+	fmt.Printf("Marmot update successful. Eris CLI Version is now: %s\n", VERSION)
+}
+
+func checkGitAndGo() {
```
Comment (compleatang) [inline commands/update.go]: let's pull this function and the next into `utils` we'll likely need them for other things down the road.

File: commands/update.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	changeDirectory()
+
+	var branch string
+	if do.Branch != "" {
+		branch = do.Branch
+	} else {
+		branch = "master"
+	}
+
+	checkoutArgs := []string{"checkout", branch}
```
Comment (compleatang) [inline commands/update.go]: may make sense to pull most of this function into utils.

File: commands/update.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	fmt.Printf("%s pulled successfully\n", branch)
+
+	goArgs := []string{"install", "./cmd/eris"}
+	out3, err := exec.Command("go", goArgs...).CombinedOutput()
+	if err != nil {
+		fmt.Printf("error with go install ./cmd/eris:\n%s\n", string(out3))
+		os.Exit(1)
+	}
+
+	fmt.Printf("Marmot update successful. Eris CLI Version is now: %s\n", VERSION)
```
Comment (compleatang) [inline commands/update.go]: run `eris init`...? (i can see args either way)

**Iff** we do run `eris init` then we'd want to add a few flags so:

* `--tool` would only update the tool (current functionality)
* `--services` would only update the default services (just the pullRepo("eris-services")) basically
* `--actions` would only update the default actions ^
* `--all` (default) would do all the above.

File: commands/init.go [hunk 4]
```diff
@@ -1,18 +1,59 @@
...
+	} else if !do.Tool {
+		do.Services = false
+		do.Actions = false
+		ini.Initialize(do)
+	}
+
+	if do.Tool {
+		util.UpdateEris(do.Branch)
+	}
+}
```
Comment (zramsay) [inline commands/init.go]: can't seem to nail the logic down here. maybe my approach is wrong?

**Comment by compleatang** [comment]
OK. I think that logic works well now. I've refactored it a bit to make it cleaner.

**Pull request merged**